### PR TITLE
Adding nascent source site plot capability.

### LINKED ISSUE #102: Plot source sites using `opnemc.lib.sample_exeternal_source`
As part of the latest NEA OpenMC workshop, we plotted source sites on top of the geometry and tally information. We did this using the `openmc.Tracks` class, but I don't see why we couldn't do the same here with the `openmc.lib.sample_external_source` feature. 

It would provide an somewhat interactive to provide users some assurance that their source is defined correctly.

### PR DESCRIPTION
**Pull request by pshriwise**
This feature is a little rough still but it adds the ability to plot source sites on top of the current view. All sites are plotted whether they're on the slice plane or not. There are a number of things I'd want to add (see below) before incorporating this, but it has a significant amount in place so I thought I'd create a draft PR at least.

See utilization in https://github.com/openmc-dev/openmc/pull/2857

TODO:
- [x] Toggle for source site visibility in the main dock
- [x] Toggle filter of visible souce sites based on coincidence with slice plane
- [ ] Arrows for source locations
- [ ] Colormap bar for particle energy

### CODE REVIEW

**Comment by shimwell** [comment]
Nice to see this PR, I remember a few people where asking for this on the slack channel a while back.

I'm keen to mirror this functionality on the python api if that is ok

**Comment by shimwell** [comment]
Perhaps a variable could be added for number of source samples (```n_samples```)

**Comment by pshriwise** [comment]
I built this out a little bit today and I'm pretty happy with it now. The animation below shows an annular `CylindricalIndependent` source with R=1 and Z=[0,10]. Main additions include the ability to toggle visibility, change the color, and filter source particles out that are too far from the slice plane.

Recording

**Comment by pshriwise** [comment]
A couple of features to be added yet I suppose, but I'd call this a good base for the feature and wouldn't mind seeing it merged as-is

**Comment by paulromano** [comment]
@pshriwise I ended up just forging ahead and implementing the changes we talked about today. Let me know if you're happy with how it looks/functions and if so, we can go ahead and merge. Thanks for adding this feature!

File: openmc_plotter/plotgui.py [hunk 1]
```diff
@@ -648,9 +649,34 @@ def updatePixmap(self):
         self.ax.dataLim.y0 = data_bounds[2]
         self.ax.dataLim.y1 = data_bounds[3]
 
+
```
Comment (pshriwise) [inline openmc_plotter/plotgui.py:652]:
```suggestion
```

Comment (pshriwise) [inline openmc_plotter/plotgui.py]:
```suggestion
```

**Comment by pshriwise** [comment]
@bohmt so you can use this once it's merged.

**Review by paulromano — APPROVED** [review]

**Pull request merged**